src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -273,6 +279,20 @@ public void setFilename(final String f) {
         return this.theData;
     }
 
+    /**
+     * Return a String version of the data. If there is already a cached version of the dataString, then return that,
+     * otherwise generate the String to be returned and cache the result
+     *
+     * @return a String representation of the data
+     */
+    @Override
+    public String dataAsString() {
+        if (this.dataString == null) {
```

Comment:
We might consider caching only if the string is under a certain size. Otherwise, if the string is too large, we always re-generate the string. This may mitigate some of the concerns over IBDO bloat.